common/tests/CMakeLists.txt

```diff
@@ -29,5 +31,14 @@ if(${run_int_tests})
         build_test_folder(gballoc_ll_mimalloc_int)
         build_test_folder(gballoc_ll_win32heap_int) 
         build_test_folder(gballoc_ll_passthrough_int)
+        
+        #this builds and runs the same test for gballoc_hl but with all the implementations of the underlying gballoc_ll.
+        #this scripts sets a variable (GBALOC_LL_IMPL) that is picked as underlying implementatio of gballoc_ll and consequentially as file name (gballoc_ll_${GBALOC_LL_IMPL})
+        foreach(GBALOC_LL_IMPL ${GBALLOC_LL_TYPE_VALUES})

```

Comment:
>GBALLOC_LL_TYPE_VALUES [](start = 33, length = 22)

where is this defined?

Reply:
This is defined in the master CMakeLists.txt of this repo. It contains all the values that the LL can have (and it also checks that the user didn't pass some unknown value). Right now this list has 3 values: "passthrough (directly to malloc) win32heap and mimalloc".

The purpose here being to check that every implementation of the HL (at the moment 1) can work with any implementation of the LL layer - composability.

---
In reply to: [458466154](https://github.com/Azure/azure-c-pal/pull/12#discussion_r458466154) [](ancestors = 458466154)